crates/teloxide/src/dispatching/dialogue/get_chat_id.rs

```diff
@@ -23,3 +26,89 @@ impl GetChatId for Update {
         self.chat().map(|chat| chat.id)
     }
 }
+
+impl GetChatId for Recipient {
+    fn chat_id(&self) -> Option<ChatId> {
+        match self {
+            Recipient::Id(chat_id) => Some(*chat_id),
+            Recipient::ChannelUsername(_) => None,
+        }
+    }
+}
+
+impl GetChatId for BotCommandScope {
+    fn chat_id(&self) -> Option<ChatId> {
+        match self {
+            BotCommandScope::Default
+            | BotCommandScope::AllPrivateChats
+            | BotCommandScope::AllGroupChats
+            | BotCommandScope::AllChatAdministrators => None,
+            BotCommandScope::Chat { chat_id: recipient }
+            | BotCommandScope::ChatAdministrators { chat_id: recipient }
+            | BotCommandScope::ChatMember { chat_id: recipient, .. } => recipient.chat_id(),
+        }
+    }
+}
+
+impl GetChatId for Chat {
+    fn chat_id(&self) -> Option<ChatId> {
+        Some(self.id)
+    }
+}
+
+impl GetChatId for UpdateKind {
```

Comment:
I don't think that it's correct. End-users don't work with UpdateKind (it's a kind of utility field, smth that works under the hood), but directly with `Message`, `InlineQuery`, `ChosenInlineResult`, etc.

So, I guess, `GetChatId` has to be implemented individually on them